apps/grades-sync/src/grade_repository.rs

```diff
@@ -0,0 +1,270 @@
+use crate::pg::Database;
+use crate::subject_repository::Subject;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use sqlx::types::Uuid;
+use sqlx::{Executor, FromRow};
+
+#[derive(Debug, PartialEq, PartialOrd, sqlx::Type, Deserialize, Serialize, Copy, Clone)]
+#[sqlx(type_name = "subject_grading_season", rename_all = "UPPERCASE")]
+pub enum SubjectGradingSeason {
+    Winter,
+    Spring,
+    Summer,
+    Autumn,
+}
+
+#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
+pub enum SubjectGradingKey {
+    A,
+    B,
+    C,
+    D,
+    E,
+    F,
+    G,
+    H,
+}
+
+impl SubjectGradingKey {
+    pub fn to_column_key_name(self) -> &'static str {
+        match self {
+            SubjectGradingKey::A => "graded_a",
+            SubjectGradingKey::B => "graded_b",
+            SubjectGradingKey::C => "graded_c",
+            SubjectGradingKey::D => "graded_d",
+            SubjectGradingKey::E => "graded_e",
+            SubjectGradingKey::F => "graded_f",
+            SubjectGradingKey::G => "graded_pass",
+            SubjectGradingKey::H => "graded_fail",
+        }
+    }
+
+    pub fn to_multiplication_factor(self) -> f32 {
+        match self {
+            SubjectGradingKey::A => 5.0,
+            SubjectGradingKey::B => 4.0,
+            SubjectGradingKey::C => 3.0,
+            SubjectGradingKey::D => 2.0,
+            SubjectGradingKey::E => 1.0,
+            SubjectGradingKey::F => 0.0,
+            x => panic!("invalid grading key: {:?}", x),
+        }
```

Comment:
```suggestion
    pub fn to_multiplication_factor(self) -> Option<f32> {
        match self {
            SubjectGradingKey::A => Some(5.0),
            SubjectGradingKey::B => Some(4.0),
            SubjectGradingKey::C => Some(3.0),
            SubjectGradingKey::D => Some(2.0),
            SubjectGradingKey::E => Some(1.0),
            SubjectGradingKey::F => Some(0.0),
            _ => None,
        }
```

Reply:
Yeah, could do. I was also wondering if I just inline it at call-site, since the call site guarantees that it's not G/H